lib/facter/metadata_retrieval.rb

```diff
@@ -0,0 +1,19 @@
+require 'net/http'
+require 'net/https'
+uri = URI.parse('http://169.254.169.254/2014-11-05/dynamic/instance-identity/document')
+http = Net::HTTP.new(uri.host, uri.port)
+http.open_timeout = 4
+http.read_timeout = 4
+begin
+  http.start
+  response = http.request(Net::HTTP::Get.new(uri.request_uri))
+rescue StandardError
+  raise 'Unable to reach AWS metadata'
```

Comment:
Perhaps print the exception that was caught (unless Ruby already automagically does this) to give a little more detail to why the metadata fetch failed.